kura/org.eclipse.kura.deployment.agent/src/main/java/org/eclipse/kura/deployment/agent/impl/DeploymentAgent.java

```diff
@@ -335,17 +337,27 @@ private void installPackagesFromConfFile() {
 
         Set<Object> packageNames = deployedPackages.keySet();
         for (Object packageName : packageNames) {
-            String packageUrl = (String) deployedPackages.get(packageName);
-
-            logger.info("Deploying package name {} at URL {}", packageName, packageUrl);
             try {
-                installDeploymentPackageAsync(packageUrl);
+                String packageUri = (String) deployedPackages.get(packageName);
+
+                if (isHttp(packageUri)) {
+                    throw new KuraRuntimeException(KuraErrorCode.SECURITY_EXCEPTION,
+                            "For security reason is not allowed install package from remote repository in config file.");
+                }
+
+                logger.info("Deploying package name {} at URI {}", packageName, packageUri);
+                installDeploymentPackageAsync(packageUri);
             } catch (Exception e) {
                 logger.error("Error installing package {}", packageName, e);
             }
         }
     }
 
+    private boolean isHttp(String packageUri) throws URISyntaxException {
```

Comment:
I would only allow file URIs, not just exclude http protocols (we leave out ftp and others). Something like:

```java
private boolean isFile(String packageUri) {
    new File(packageUri).isFile();
}
```